File: sort/cpp/insertion_sort.cpp
#include <iostream>
#include "helper.h"
#include "sort_functions.h"
using namespace std;

int main()
{
    std::ios::sync_with_stdio(false); // for faster input and output using cin and cout
    int size;
    cout << "Enter size of array: ";
    cin >> size;
    int arr[size];
    cout << "\nEnter array elements: ";
    for(int i=0; i<size; i++)
    {
        cin >> arr[i];
    }
    
    insertion_sort(arr, size);
    cout << "Array after sorting is: ";
    printArray(arr, size);

return 0;
}

File: sort/python3/insertion_sort.py
size = int(input("Enter size of your array: "))
arr = []
element = input("Enter array elements: ").split(' ')
for i in range(0, size):
    arr.append(int(element[i]))

for j in range(1, size):
    key = arr[j]
    i = j-1
    while i>=0 and arr[i]>key:
        arr[i+1]=arr[i]
        i=i-1
    arr[i+1]=key    
        
print("After insertion sort, array is: ", arr)
